dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -227,18 +228,26 @@ public FunctionView Describe()
     /// <inheritdoc/>
     public async Task<SKContext> InvokeAsync(SKContext context, CompleteRequestSettings? settings = null, CancellationToken cancellationToken = default)
     {
+        using var activity = s_activitySource.StartActivity($"{this.SkillName}.{this.Name}");
+        this.s_executionTotalCounter.Add(1);
+        context.Log.LogInformation("{SkillName}.{StepName}: Function execution started", this.SkillName, this.Name);
+        var stopwatch = new Stopwatch();
+        stopwatch.Start();
```

Comment:
I think telemetry-related logic should be moved to separate place to keep `InvokeAsync` method clean and readable.

Reply:
Agree, I moved it both in `Plan` and `SKFunction` to different method called `InvokeAsyncInstrumented` as you did before.